Spec Account management for OAuth 2.0 API (#2270)
matrix-org/matrix-spec: content/client-server-api/_index.md

- For other errors, the server returns a `400 Bad Request` response with error
  details

#### Account management {id="oauth-20-account-management"}

{{% added-in v="1.18" %}}

All account management is done via the homeserver's web UI.

This specification defines extensions to the [OAuth Authorization Server
Metadata registry](https://www.iana.org/assignments/oauth-parameters/oauth-parameters.xhtml#authorization-server-metadata)
to offer clients a way to deep-link to the account management capabilities of
the homeserver to allow the user to complete the account management operations
in a browser.

##### Account management URL discovery

The [OAuth 2.0 authorization server metadata](#server-metadata-discovery) is
extended to include the following fields:

| Field                                  | Description                                                                                     |
|----------------------------------------|-------------------------------------------------------------------------------------------------|
| `account_management_uri`               | The URL where the user is able to access the account management capabilities of the homeserver. |

> | `account_management_actions_supported` | An array of actions that the account management URL supports, as defined below.                 |

Reviewer: iirc we have a way to render arbitrary schemas - we should use that here to meet the specification requirements more easily (namely types).

##### Account management URL parameters

> The account management URL MAY accept the following query parameters:

Reviewer: ```suggestion
The account management URL MAY accept the following minimum query parameters:
```

- For other errors, the server returns a `400 Bad Request` response with error
  details

#### Account management {id="oauth-20-account-management"}

{{% added-in v="1.18" %}}

> All account management is done via the homeserver’s web UI as all endpoints that

Reviewer 2: ```suggestion
All account management is done via the homeserver's web UI as all endpoints that
```

require User-Interactive Authentication are unsupported by this authentication

> API.

Author: Removed in edea11e61010b7e670fca3bad95be41b45d8b9f4

advertised as supported by the server then the server SHOULD support the
`device_id` parameter.

##### Account management URL actions

The following account management actions are defined:

| Action                           | Description                                                                                                                                          |
|----------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------|
| `org.matrix.profile`             | The user wishes to view/edit their profile (name, avatar, contact details).                                                                          |
| `org.matrix.devices_list`        | The user wishes to view a list of their devices.                                                                                                     |
| `org.matrix.device_view`         | The user wishes to view the details of a specific device. A `device_id` should be provided.                                                          |

> | `org.matrix.device_delete`       | The user wishes to delete/log out a specific device. A `device_id` should be provided.                                                               |

Reviewer 2: ```suggestion
| `org.matrix.device_view`         | The user wishes to view the details of a specific device. A `device_id` SHOULD be provided.                                                          |
| `org.matrix.device_delete`       | The user wishes to delete/log out a specific device. A `device_id` SHOULD be provided.                                                               |
```
